src/data/dfinsupp/basic.lean

```diff
@@ -148,6 +148,14 @@ end⟩
   zip_with f hf g₁ g₂ i = f i (g₁ i) (g₂ i) :=
 rfl
 
+/-- Merge two finitely supported functions `x y : Π₀ i, α i`; at every coordinate `a : α`, use the
+  predicate `p` to decide whether to take the value of `x` or the value of `y`. -/
+noncomputable def merge (p : ι → Prop) (x y : Π₀ i, β i) : Π₀ i, β i :=
```

Comment:
To me this looks like it should be called `piecewise` to match `set.piecewise`; likely meaning that `p` should be replaced with a set.

Reply:
Thanks! I changed the name to `dfinsupp.piecewise` and made it computable like `set.piecewise`. The added decidability argument in `piecewise` requires me to use `classical` in the proof of `lex.acc_of_single_erase`, and a bit unfortunately it conflicts with the decidability instance for the lemma statement, so I have to an extra `convert`s in the proof now.